include/fluent-bit/flb_event_loop.h

```diff
@@ -71,32 +71,27 @@ static inline void flb_event_load_injected_events(struct flb_bucket_queue *bktq,
-                                                                                        \
-        /* condition */                                                                 \
-        event != NULL &&                                                                \
-        (__flb_event_priority_live_foreach_iter < max_iter || max_iter == -1);          \
-                                                                                        \
-        /* update */                                                                    \
-        ++__flb_event_priority_live_foreach_iter,                                       \
-        flb_event_load_injected_events(bktq, evl,                                       \
-                                      __flb_event_priority_live_foreach_n_events),      \
-        mk_event_wait_2(evl, 0),                                                        \
-        __flb_event_priority_live_foreach_n_events = evl->n_events,                     \
-        flb_event_load_bucket_queue(bktq, evl),                                         \
-        event = flb_bucket_queue_find_min(bktq) ?                                       \
-                mk_list_entry(                                                          \
-                    flb_bucket_queue_pop_min(bktq), struct mk_event, _priority_head) :  \
-                NULL                                                                    \
+#define flb_event_priority_live_foreach(event, bktq, evl, max_iter)                                   \
+    int __flb_event_priority_live_foreach_iter = 0;                                                   \
+    int __flb_event_priority_live_foreach_n_events = evl->n_events;                                   \
+    for (                                                                                             \
+        /* init */                                                                                    \
+        flb_event_load_bucket_queue(bktq, evl);                                                       \
+                                                                                                      \
+        /* condition */                                                                               \
+        (__flb_event_priority_live_foreach_iter < max_iter || max_iter == -1) &&                      \
+        (NULL != (                                                                                    \
+            event = flb_bucket_queue_find_min(bktq)                                                   \
+                    ? mk_list_entry(flb_bucket_queue_pop_min(bktq), struct mk_event, _priority_head)  \
+                    : NULL                                                                            \
+        ));                                                                                           \
```

Comment:
@matthewfala Thank you for taking the time to check this.

> Regardless, it's interesting how you are seeing lost events. If the event is not processed, I would think it would be re-added to the bucket queue on the next pass because the event is technically still triggered/in the ready list.
> 
> But then again, if it keeps happening due to a situation where max_iter events are repeatedly queued before the problematic event, it might keep being dropped and picked up by the loop. Or it could be that event is some kind of one-shot event, which I don't think fluent bit uses currently.

That's the thing – I don't think the event is re-added to the bucket queue on the next pass. There are two ways the event is added: 
* `flb_event_load_bucket_queue` – doesn't work for the timed out event, as its status is `MK_EVENT_NONE`
* `flb_event_load_injected_events` – this only adds the injected event once, when the injected event is new in the event loop, and event loop count > `__flb_event_priority_live_foreach_n_events` loop counter variable; this variable is then updated and this injected event is skipped next time

> I think your change should be accepted as long as it has been tested thoroughly!

I've been testing this internally at my employer; not yet rolled out to production. I think it would be nice to have a test for this, so I'll try to add one.

> A small style note. There's some fluent bit max line columns limit in the style guide, I forget what it is, but I think the changes may have gone over the style limits. Maybe check to see if the lines are less than whatever that column limit is.

Thanks for the note. I will check the style guide, and ask Eduardo and team to review.